modules/core/src/main/scala/org/scalasteward/core/util/string.scala

```diff
@@ -70,7 +72,29 @@ object string {
   ): Option[String] = {
     var changed = false
     val replacer1 = replacer.andThen(_.map(r => { changed = true; r }))
-    val result = regex.replaceSomeIn(target, replacer1)
+    val targetString = target.toString
+    val result = if (!targetString.contains("scala-steward:off")) {
+      regex.replaceSomeIn(target, replacer1)
+    } else {
+      var ignoreLines = false
+      targetString.linesIterator
+        .map(s => {
+          if (ignoreLines) {
+            if (s.contains("scala-steward:on")) {
+              ignoreLines = false
+              regex.replaceSomeIn(s, replacer1)
```

Comment:
One concern I have is that the regular expressions now operate on single lines only. Before this change they operated on the whole content of files.

Reply:
What is your worry?
I don't think operating on every single lines separately could be a problem
Also there is a fast-path (for most repositories), where the whole content is replaced one time if no `scala-steward:off` found.

Reply:
You're right that our current regex is line based so this is not a breaking change. But what if we want to use a regex that works on multiple lines in the future. Those wouldn't work in files that contain `scala-steward:off`.

What do you think about this: We split the content of a file into multiple parts where each part is associated with a `Boolean` that specifies if that part can be changed by Scala Steward. This would have a signature similar to this: `String => List[(String, Boolean)]`. We would then only apply our replace function on those parts where the `Boolean` is `true` (where we are allowed to do changes).

I have a feeling that this approach would also be more compositional.

Reply:
Refactored so in https://github.com/fthomas/scala-steward/pull/671/commits/4f28b79fbb89d1deaeb0333a1a983575e182c9bd